hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenDissociateFromAccountHandler.java

```diff
@@ -125,7 +125,10 @@ public void handle(@NonNull final HandleContext context) {
                 }
             } else {
                 // Handle active tokens
-                validateFalse(tokenRel.accountNumber() == token.treasuryAccountNumber(), ACCOUNT_IS_TREASURY);
+                validateFalse(
+                        token.treasuryAccountId() != null
+                                && tokenRel.accountId().equals(token.treasuryAccountId()),
```

Comment:
for null safety, invert this:
`token.treasuryAccountId().equals(tokenRel.accountId())`